src/Nethermind/Ethereum.Test.Base/BlockchainTestBase.cs

```diff
@@ -104,19 +103,13 @@ protected async Task<EthereumTestResult> RunTest(BlockchainTest test, Stopwatch?
                 Assert.Fail("Expected genesis spec to be Frontier for blockchain tests");
             }
 
-            bool isNetworkLondon = test.NetworkAfterTransition == London.Instance || test.Network == London.Instance;
-            bool isNetworkGrayGlacier = test.NetworkAfterTransition == GrayGlacier.Instance || test.Network == GrayGlacier.Instance;
-            bool isNetworkShanghai = test.NetworkAfterTransition == Shanghai.Instance || test.Network == Shanghai.Instance;
-            bool isNetworkCancun = test.NetworkAfterTransition == Cancun.Instance || test.Network == Cancun.Instance;
-
-            if (isNetworkLondon || isNetworkGrayGlacier || isNetworkShanghai || isNetworkCancun)
-            {
-                // TODO: Skip test. Will fix in the next pr
-                return new EthereumTestResult(test.Name, null, true);
-            }
-
             DifficultyCalculator.Wrapped = new EthashDifficultyCalculator(specProvider);
             IRewardCalculator rewardCalculator = new RewardCalculator(specProvider);
+            if (test.Network == GrayGlacier.Instance || test.Network == Shanghai.Instance || test.Network == Cancun.Instance)
```

Comment:
Maybe just add IsPostMerge property?

Reply:
And `Shanghai : GrayGlacier`, I think we are missing Paris there!

Reply:
@LukaszRozmej you mean to IReleaseSpec?

Reply:
Yes ReleaseSpec type hierarhy.

Reply:
If we add `ReleaseSpec.IsPostMerge` we will not be able to get spec with `ForkActivation`(block number and timestamp, not ttd). Maybe I don't see workaround here. And I think adding this functionality not worth it
@LukaszRozmej @MarekM25 wdyt?

Reply:
Why we won't able to get it?

Reply:
To get block spec we use this function: `IReleaseSpec GetSpec(ForkActivation forkActivation)`. To understand if this block is post merge we have to add ttd parameter to `ForkActivation`. As I understand merge happened between GrayGlacier and Shanghai. So we will have to add one more `ReleaseSpec`.

Reply:
Can we leverage `ISpecProvider.MergeBlockNumber`? We already have it.

Reply:
We might not have it before transition. I'm afraid that it might be broken in some situations. And I think adding it to fix one if in test integration is overkill.

Reply:
It will probably work if we ensure that we never get block's SpecProvider before setting MergeBlockNumber. Or maybe it will just work. Too complicated imo

Reply:
kiss:)